pkg/registry/servicecatalog/broker/validation.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
I would personally rather put validation code in pkg/apis/servicecatalog/validation, but I can understand why you would want to place them under registry.

One motivation for keeping them outside the registry package in k8s is that there are places where some objects are validated in system components that are not the API server.  For example, the kubelet can accept files as a source of pods and has its own HTTP endpoint in addition to the API server, so the kubelet itself applies validations.

I don't think we need to export any validations here, but I do think it would be best to group them under API instead of registry, since that is where people with k8s experience will look for them.  I realize that isn't a convincing argument to many people in the SIG, so consider this: if you are working on both k8s and service-catalog, do you want to have to look in fundamentally different places for validation code?  I pose that as a sincere question.

For the record, I also view validations as being more closely related to the object definitions themselves than the details of how those objects are stored.